docs/upgrade-to-v3.md

```diff
@@ -33,6 +33,136 @@ The Microsoft Graph Java SDK now leverages azure-identity to improve the authent
 1. Remove any dependency on `microsoft-graph-android-auth` and replace them by one of the providers available in azure-identity.
 1. Reconsider the need for custom authentication providers (implementation of `IAuthenticationProvider` or `ICoreAuthenticationProvider`) that might be part of your project and use a provider available in azure-identity whenever possible.
 
+#### Authentication upgrade examples
```

Comment:
It would be helpful to move these examples into a separate doc say TokenCredentialSamples.md. It would be easier to find this. I think of the upgrade doc as more of a change log.

Reply:
thanks for the suggestion. updated.